File: .sdk-repos/version-8.9/orchestration-cluster-api-python/examples/group.py
# Compilable usage examples for group management operations.
# These examples are type-checked during build to guard against API regressions.
from __future__ import annotations

from camunda_orchestration_sdk import (
    CamundaClient,
    GroupCreateRequest,
    GroupSearchQueryRequest,
    GroupUpdateRequest,
    MappingRuleSearchQueryRequest,
    RoleSearchQueryRequest,
    Unset,
    Username,
)


# region CreateGroup
def create_group_example() -> None:
    client = CamundaClient()

    result = client.create_group(
        data=GroupCreateRequest(group_id="engineering", name="Engineering"),
    )

    print(f"Group: {result.group_id}")
# endregion CreateGroup


# region GetGroup
def get_group_example() -> None:
    client = CamundaClient()

    result = client.get_group(group_id="engineering")

    print(f"Group: {result.name}")
# endregion GetGroup


# region SearchGroups
def search_groups_example() -> None:
    client = CamundaClient()

    result = client.search_groups(
        data=GroupSearchQueryRequest(),
    )

    if not isinstance(result.items, Unset):
        for group in result.items:
            print(f"Group: {group.name}")
# endregion SearchGroups


# region UpdateGroup
def update_group_example() -> None:
    client = CamundaClient()

    client.update_group(
        group_id="engineering",
        data=GroupUpdateRequest(name="engineering-team"),
    )
# endregion UpdateGroup


# region DeleteGroup
def delete_group_example() -> None:
    client = CamundaClient()

    client.delete_group(group_id="engineering")
# endregion DeleteGroup


# region AssignUserToGroup
def assign_user_to_group_example(username: Username) -> None:
    client = CamundaClient()

    client.assign_user_to_group(
        group_id="engineering",
        username=username,
    )
# endregion AssignUserToGroup


# region UnassignUserFromGroup
def unassign_user_from_group_example(username: Username) -> None:
    client = CamundaClient()

    client.unassign_user_from_group(
        group_id="engineering",
        username=username,
    )
# endregion UnassignUserFromGroup


# region AssignClientToGroup
def assign_client_to_group_example() -> None:
    client = CamundaClient()

    client.assign_client_to_group(
        group_id="engineering",
        client_id="my-service-account",
    )
# endregion AssignClientToGroup


# region UnassignClientFromGroup
def unassign_client_from_group_example() -> None:
    client = CamundaClient()

    client.unassign_client_from_group(
        group_id="engineering",
        client_id="my-service-account",
    )
# endregion UnassignClientFromGroup


# region AssignMappingRuleToGroup
def assign_mapping_rule_to_group_example() -> None:
    client = CamundaClient()

    client.assign_mapping_rule_to_group(
        group_id="engineering",
        mapping_rule_id="rule-123",
    )
# endregion AssignMappingRuleToGroup


# region UnassignMappingRuleFromGroup
def unassign_mapping_rule_from_group_example() -> None:
    client = CamundaClient()

    client.unassign_mapping_rule_from_group(
        group_id="engineering",
        mapping_rule_id="rule-123",
    )
# endregion UnassignMappingRuleFromGroup


# region SearchUsersForGroup
def search_users_for_group_example() -> None:
    client = CamundaClient()

    result = client.search_users_for_group(
        group_id="engineering",
    )

    if not isinstance(result.items, Unset):
        for user in result.items:
            print(f"User: {user.username}")
# endregion SearchUsersForGroup


# region SearchClientsForGroup
def search_clients_for_group_example() -> None:
    client = CamundaClient()

    result = client.search_clients_for_group(
        group_id="engineering",
    )

    if not isinstance(result.items, Unset):
        for c in result.items:
            print(f"Client: {c.client_id}")
# endregion SearchClientsForGroup


# region SearchRolesForGroup
def search_roles_for_group_example() -> None:
    client = CamundaClient()

    result = client.search_roles_for_group(
        group_id="engineering",
        data=RoleSearchQueryRequest(),
    )

    if not isinstance(result.items, Unset):
        for role in result.items:
            print(f"Role: {role.name}")
# endregion SearchRolesForGroup


# region SearchMappingRulesForGroup
def search_mapping_rules_for_group_example() -> None:
    client = CamundaClient()

    result = client.search_mapping_rules_for_group(
        group_id="engineering",
        data=MappingRuleSearchQueryRequest(),
    )

    if not isinstance(result.items, Unset):
        for rule in result.items:
            print(f"Mapping rule: {rule.mapping_rule_id}")
# endregion SearchMappingRulesForGroup

File: .sdk-repos/version-8.9/orchestration-cluster-api-csharp/examples/Group.cs
// Compilable usage examples for group management operations.
// These examples are type-checked during build to guard against API regressions.
using Camunda.Orchestration.Sdk;

public static class GroupExamples
{
    #region CreateGroup
    // <CreateGroup>
    public static async Task CreateGroupExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.CreateGroupAsync(new GroupCreateRequest
        {
            GroupId = "engineering",
            Name = "Engineering",
        });

        Console.WriteLine($"Group key: {result.GroupId}");
    }
    // </CreateGroup>
    #endregion CreateGroup

    #region GetGroup

    // <GetGroup>
    public static async Task GetGroupExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.GetGroupAsync("engineering");
        Console.WriteLine($"Group: {result.Name}");
    }
    // </GetGroup>
    #endregion GetGroup

    #region SearchGroups

    // <SearchGroups>
    public static async Task SearchGroupsExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchGroupsAsync(new GroupSearchQueryRequest());

        foreach (var group in result.Items)
        {
            Console.WriteLine($"Group: {group.Name}");
        }
    }
    // </SearchGroups>
    #endregion SearchGroups

    #region UpdateGroup

    // <UpdateGroup>
    public static async Task UpdateGroupExample()
    {
        using var client = CamundaClient.Create();

        await client.UpdateGroupAsync("engineering", new GroupUpdateRequest
        {
            Name = "engineering-team",
        });
    }
    // </UpdateGroup>
    #endregion UpdateGroup

    #region DeleteGroup

    // <DeleteGroup>
    public static async Task DeleteGroupExample()
    {
        using var client = CamundaClient.Create();

        await client.DeleteGroupAsync("engineering");
    }
    // </DeleteGroup>
    #endregion DeleteGroup

    #region AssignUserToGroup

    // <AssignUserToGroup>
    public static async Task AssignUserToGroupExample(Username username)
    {
        using var client = CamundaClient.Create();

        await client.AssignUserToGroupAsync("engineering", username);
    }
    // </AssignUserToGroup>
    #endregion AssignUserToGroup

    #region UnassignUserFromGroup

    // <UnassignUserFromGroup>
    public static async Task UnassignUserFromGroupExample(Username username)
    {
        using var client = CamundaClient.Create();

        await client.UnassignUserFromGroupAsync("engineering", username);
    }
    // </UnassignUserFromGroup>
    #endregion UnassignUserFromGroup

    #region AssignClientToGroup

    // <AssignClientToGroup>
    public static async Task AssignClientToGroupExample()
    {
        using var client = CamundaClient.Create();

        await client.AssignClientToGroupAsync("engineering", "my-service-account");
    }
    // </AssignClientToGroup>
    #endregion AssignClientToGroup

    #region UnassignClientFromGroup

    // <UnassignClientFromGroup>
    public static async Task UnassignClientFromGroupExample()
    {
        using var client = CamundaClient.Create();

        await client.UnassignClientFromGroupAsync("engineering", "my-service-account");
    }
    // </UnassignClientFromGroup>
    #endregion UnassignClientFromGroup

    #region AssignMappingRuleToGroup

    // <AssignMappingRuleToGroup>
    public static async Task AssignMappingRuleToGroupExample()
    {
        using var client = CamundaClient.Create();

        await client.AssignMappingRuleToGroupAsync("engineering", "rule-123");
    }
    // </AssignMappingRuleToGroup>
    #endregion AssignMappingRuleToGroup

    #region UnassignMappingRuleFromGroup

    // <UnassignMappingRuleFromGroup>
    public static async Task UnassignMappingRuleFromGroupExample()
    {
        using var client = CamundaClient.Create();

        await client.UnassignMappingRuleFromGroupAsync("engineering", "rule-123");
    }
    // </UnassignMappingRuleFromGroup>
    #endregion UnassignMappingRuleFromGroup

    #region SearchUsersForGroup

    // <SearchUsersForGroup>
    public static async Task SearchUsersForGroupExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchUsersForGroupAsync(
            "engineering",
            new SearchUsersForGroupRequest());

        foreach (var user in result.Items)
        {
            Console.WriteLine($"User: {user.Username}");
        }
    }
    // </SearchUsersForGroup>
    #endregion SearchUsersForGroup

    #region SearchClientsForGroup

    // <SearchClientsForGroup>
    public static async Task SearchClientsForGroupExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchClientsForGroupAsync(
            "engineering",
            new SearchClientsForGroupRequest());

        foreach (var c in result.Items)
        {
            Console.WriteLine($"Client: {c.ClientId}");
        }
    }
    // </SearchClientsForGroup>
    #endregion SearchClientsForGroup

    #region SearchRolesForGroup

    // <SearchRolesForGroup>
    public static async Task SearchRolesForGroupExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchRolesForGroupAsync(
            "engineering",
            new RoleSearchQueryRequest());

        foreach (var role in result.Items)
        {
            Console.WriteLine($"Role: {role.Name}");
        }
    }
    // </SearchRolesForGroup>
    #endregion SearchRolesForGroup

    #region SearchMappingRulesForGroup

    // <SearchMappingRulesForGroup>
    public static async Task SearchMappingRulesForGroupExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchMappingRulesForGroupAsync(
            "engineering",
            new MappingRuleSearchQueryRequest());

        foreach (var rule in result.Items)
        {
            Console.WriteLine($"Mapping rule: {rule.MappingRuleId}");
        }
    }
    // </SearchMappingRulesForGroup>
    #endregion SearchMappingRulesForGroup
}
